app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -90,6 +93,13 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
         serviceLocator.intentLiveData.value = Consumable.from(intentData)
     }
 
+    override fun onDestroy() {
+        super.onDestroy()
+
+        // This removes the Sessions from the SessionManager so should be called last.
+        webRenderComponents.applicationLifecycleSessionCache.onDestroyCacheSessions()
+    }
```

Comment:
Are there any potential edge cases here if the `Activity` is destroyed without hitting `onDestroy`?

Reply:
If the Activity is destroyed without onDestroy getting called but the process is not destroyed then yes, the memory leak persists. After doing some reading, I was under the impression that this can't happen but the Android docs aren't explicit enough on it that I'd trust them 100%.

The original implementation was in `onStart/onStop` and also `onCreate` (in the case the activity is recreated, we need to add it before we try to use it) but it was fairly complex (e.g. what happens if someone tries to access the session after onStop?) so I prefer the simplicity of this but I understand it might not always be correct.

I'll see if we can add telemetry to detect the case where onCreate is called but the leak still exists.

Reply:
@Baron-Severin I implemented a Sentry error when this happens: please review the new commit.